tracing/decorators.py

```diff
@@ -100,18 +95,44 @@ def _execute_tracing(
             apply_opts=True,
         )
 
-        _trace_callable(tracer, lambda: c(*args, **kwargs))
+        err = _trace_callable(tracer, lambda: c(*args, **kwargs))
 
         traced = tracer.trace_data
 
     if benchmarks is not None:
```

Comment:
There is already an if benchmark.performance. Move the code 103 - 115 to the previous if.